sage-spkg-info.py

```diff
@@ -0,0 +1,413 @@
+        "--log-level", dest="log_level", default="WARNING",
+        choices=["CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG"],
+        help="Set logging verbosity (default: WARNING)",
+    )
+    return parser.parse_args()
+
+
+# To run the Helpers
+
+def run(cmd: Iterable[str], *, check: bool = True, text: bool = True) -> subprocess.CompletedProcess:
+    # To Run the subprocess and log on DEBUG.
+    logger.debug("Running command: %s", " ".join(map(shlex.quote, cmd)))
+    return subprocess.run(cmd, capture_output=True, text=text, check=check)
+
+
+# For the purpose of better looking, Output formatting helpers 
+# Splitted the helper into two small classes, RstFormatter and PlainFormatter
+class _BaseFormatter(ABC):
+    """Common API For fixing output behavior."""
+    rst: bool = False
+
+    # To default for plain text
+    def ref(self, x: str) -> str: return x
+    def spkg(self, x: str) -> str: return x
+    def issue(self, x: str) -> str: return f"https://github.com/sagemath/sage/issues/{x}"
+    def code(self, *args: str) -> str: return " ".join(args)
+    def tab(self, x: str) -> str: return f"{x}:"
+    def anchor(self, name: str) -> str: return ""     # optional helper
+
+class PlainFormatter(_BaseFormatter):
+    rst = False
+    # Did not change default behavior.
+
+class RstFormatter(_BaseFormatter):
+    rst = True
+    def ref(self, x: str) -> str: return f":ref:`{x}`"
+    def spkg(self, x: str) -> str: return self.ref(f"spkg_{x}")
+    def issue(self, x: str) -> str: return f":issue:`{x}`"
+    def code(self, *args: str) -> str: return f"``{' '.join(args)}``"
+    def tab(self, x: str) -> str: return f".. tab:: {x}"
+    def anchor(self, name: str) -> str: return f".. _{name}:\n"
+
+def make_formatter(output_rst: bool) -> _BaseFormatter:
+    return RstFormatter() if output_rst else PlainFormatter()
+
+# Core functionality
+# To Read and post-process SPKG.rst (README).
+def process_spkg_file(pkg_base: str, pkg_scripts: str, fmt, out) -> None:
+    spkg_file = os.path.join(pkg_scripts, "SPKG.rst")
+    if not os.path.isfile(spkg_file):
+        logger.info("No SPKG.rst for %s", pkg_base)
+        return
+
+    with io.open(spkg_file, "r", encoding="utf-8") as fh:
+        content = fh.read()
+    # Tweaks to make identical to the bash script 
+    content = re.sub(r'https://github\.com/sagemath/sage/issues/(\d+)',
+                     r':issue:`\1`', content)
+    content = re.sub(r'https://arxiv\.org/abs/cs/(\d+)',
+                     r':arxiv:`cs/\1`', content)
+    print(content, file=out)
+
+
+def display_dependencies(pkg_base: str, fmt, out) -> None:
+    try:
+        result = run(["sage-package", "dependencies", "--format=plain", pkg_base],
+                     check=True)
+    except subprocess.CalledProcessError as e:
+        logger.warning("Could not get dependencies for %s: %s", pkg_base, e.stderr.strip())
+        return
+
+    print("Dependencies", file=out)
+    print("------------", file=out)
```

Comment:
I think after these section lines, a newline needs to be emitted, check with the original please